3.analyze_data/UMAP_analysis/nbconverted/2.jump_UMAP.py

```diff
@@ -0,0 +1,389 @@
+
+# In[7]:
+
+
+# Find all columns that start with Metadata
+metadata_cols = [col for col in merged_prob_df.columns if col.startswith("Metadata")]
+
+# Create a new DataFrame by selecting only "Metadata_Predicted_Class" and all other columns
+jump_df = merged_prob_df[
+    ["Metadata_Predicted_Class", "Metadata_Phenotypic_Value"]
+    + [col for col in merged_prob_df.columns if col not in metadata_cols]
+]
+
+# Add a data name column to separate between datasets
+jump_df.insert(0, "Metadata_data_name", "jump")
+
+print(jump_df.shape)
+jump_df.head(2)
+
+
+# ### Load in Mitocheck labeled data and update CellProfiler columns to match naming for JUMP
+
+# In[8]:
+
+
+# Load in labeled mitocheck data
+label_df = pd.read_csv(label_data_path)
+
+# Extract feature columns from the JUMP data filtered by phenotypic profiling model features
+feature_cols = [col for col in label_df.columns if col.startswith("CP__")]
+metadata_cols = [col for col in label_df.columns if col.startswith("Metadata_")]
+
+# Filter df with only CP features and the metadata columns
+mito_cp_df = label_df[
+    ["Mitocheck_Phenotypic_Class"]
+    + ["Cell_UUID"]
+    + ["Location_Center_X"]
+    + ["Location_Center_Y"]
+    + metadata_cols
+    + feature_cols
+]
+
+# change prefix for columns to match JUMP
+mito_cp_df.columns = mito_cp_df.columns.str.replace("CP__", "Nuclei_")
+
+print(mito_cp_df.shape)
+mito_cp_df.head()
+
+
+# ### Filter the mitocheck data
+
+# In[9]:
+
+
+# Load in labeled mitocheck data
+label_df = pd.read_csv(label_data_path)
+
+# Extract feature columns from the JUMP data filtered by phenotypic profiling model features
+feature_cols = [col for col in label_df.columns if col.startswith("CP__")]
+
+# Filter df with only CP features and the metadata column
+mito_cp_df = label_df[["Mitocheck_Phenotypic_Class"] + feature_cols]
+
+# change prefix for columns to match JUMP
+mito_cp_df.columns = mito_cp_df.columns.str.replace("CP__", "Nuclei_")
+
+# add data name column to separate between datasets
+mito_cp_df.insert(0, "Metadata_data_name", "mitocheck")
+
+# add data name column to separate between datasets
+mito_cp_df.insert(2, "Metadata_Phenotypic_Value", 1)
+
+# rename phenotypic class column from mito to match JUMP
+mito_cp_df.rename(
+    columns={"Mitocheck_Phenotypic_Class": "Metadata_Predicted_Class"}, inplace=True
+)
+
+# Find columns that are in the Mitocheck data but not the JUMP data (nuclei features only)
+diff_columns = (
+    pd.Index([col for col in mito_cp_df.columns if col.startswith("Nuclei_")])
+    .difference(merged_prob_df.columns)
+    .tolist()
+)
+
+# drop features that are not seen in JUMP to avoid merging errors
+mito_cp_df = mito_cp_df.drop(columns=diff_columns)
+
+print(mito_cp_df.shape)
+mito_cp_df.head(2)
+
+
+# ## Concat mitocheck and jump data (all nuclei features)
+
+# In[10]:
+
+
+# Concatenate the two DataFrames vertically
+merged_mito_jump_df = pd.concat([mito_cp_df, jump_df], axis=0)
```

Comment:
Is there a way to confirm that the two datasets have the same column order? Will concat respect order?

One approach to confirm this is to find the overlapping columns and then call `.reindex(columns=common_columns)` (or something like this) to confirm column order.

Reply:
This definitely would respect column order to the first data frame. I know that CellProfiler outputs will always put the columns in alphabetical order (based on feature) and I can see that the order of the JUMP data and Mito data were in the same order.

I also double checked with this line `mito_cp_df.columns.tolist() == jump_df.columns.tolist()` and they are the same order (True).